### PR TITLE
Support documented mysql promise based insert command.

### PR DESCRIPTION
**Pull request by rgrwkmn**
This updates the default DB insert command to match the documentation. Previously it seems that it was not updated to support promises like other DB commands were.

[Mysql insert documentation](https://db-migrate.readthedocs.io/en/latest/API/SQL/#inserttablename-columnnamearray-valuearray-callback)

It doesn't note that the callback is optional, but in the [Creating Migrations](https://db-migrate.readthedocs.io/en/latest/Getting%20Started/usage/#creating-migrations) documentation it states that db-migrate upgraded to a promise-based API, and in [other methods](https://db-migrate.readthedocs.io/en/latest/API/programable/#createdatabasedbname-callback) it states that the callback function is omitted if you wish to use the promise API. So that is how I expect all of the methods to work.

I'm making as few changes to the code as possible because this function is passed through directly by the mysql driver, called with 3 args in the [nosql driver](https://github.com/db-migrate/mongodb/blob/master/index.js#L189), and maybe more, so I'm trying to keep it backwards compatible.

**tl:dr;** Lots of justification for a really small change.

Note that the undocumented `update` method has a similar problem which I haven't fixed.

### CODE REVIEW

**Comment by wzrdtales** [comment]
Thank you for your contribution! :tada: 

Please make sure to comply to the commit standards. Your commit needs to formatted in the angular convention. This allows us to automatically generate changelogs. Thank you.

For more information please read this document: https://github.com/conventional-changelog/conventional-changelog/tree/8fb86d39cc9bf9d54e4bf5135a87b15a7c02dc25/packages/conventional-changelog-angular

**Comment by rgrwkmn** [comment]
Thanks for calling that out, I couldn't tell what commitlint wanted me to do.

**Comment by wzrdtales** [comment]
When you click on details in the checks session, you get directed to the readme, this probably needs to be in the message as well. I will take care of that :)

File: index.js [hunk 1]
```diff
@@ -340,11 +340,11 @@ var Base = Class.extend({
 
     var columnNameArray = {};
 
-    if( arguments.length > 3 ) {
+    if( arguments.length > 3 || Array.isArray(callback)) {
 
       columnNameArray = valueArray;
       valueArray = callback;
-      callback = arguments[3];
+      callback = arguments[3] ? arguments[3] : throw;
```
Comment (wzrdtales) [inline index.js]: What should this line accomplish? I don't think it is a good idea to put in throw on missing callbacks. That will definitely collide with promises.

**Review by wzrdtales — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (rgrwkmn) [reply index.js]: Ah, I see, it puts the callback into nodeify at the end. In that case I can check that callback is defined before trying to use it on line 336. Should I amend this commit or create a new one?

Comment (wzrdtales) [reply index.js]: as you want, we don't enforce squashing though.

File: index.js [hunk 2]
```diff
@@ -363,7 +362,12 @@ var Base = Class.extend({
     }
 
     if (columnNameArray.length !== valueArray.length) {
-      return callback(new Error('The number of columns does not match the number of values.'));
+      var error = new Error('The number of columns does not match the number of values.');
+      if (typeof callback === 'function') {
+        return callback(error);
+      } else {
+        throw error;
```
Comment (wzrdtales) [inline index.js:369]: it is better to return Promise.reject(error), I will replace that afterwards myself, will merge this now.

**Review by wzrdtales — APPROVED** [review]

**Comment by wzrdtales** [comment]
Here is the adjustment for reference https://github.com/db-migrate/db-migrate-base/commit/77bcc68633655f3868b8eaccc47b0262650791a9#diff-168726dbe96b3ce427e7fedce31bb0bcR401

**Comment by rgrwkmn** [comment]
Cool, thanks!
